src/test/scala/PSO/ParticleSpec.scala

```diff
@@ -21,14 +21,14 @@ import org.scalatest.BeforeAndAfterAll
 import org.scalatest.WordSpecLike
 import org.scalatest.Matchers
 
-import com.trifectalabs.myriad.ProblemType
-import com.trifectalabs.myriad.pso.{ PSOConfiguration, PSOSystemFactory,
-  ParticleTopology, ReportRequest, Report, ComputeIteration }
-import com.typesafe.config.ConfigFactory
+import com.trifectalabs.myriad.{ParticleState, SwarmMaster, SampleFunctions,
+  ProblemType}
+import com.trifectalabs.myriad.pso._
 
-import akka.actor.ActorSystem
+import akka.actor.{Props, ActorSystem}
 import akka.testkit.{ DefaultTimeout, ImplicitSender, TestKit }
 import scala.concurrent.duration._
+import akka.testkit.TestActorRef
 
 class ParticleSpec
```

Comment:
http://scalamock.org/user-guide/mocking_style/#expectations-first-style

Essentially, the *unit* test will test a function or service method. The function will perform some operation, and/or call other methods. We test this function by ensuring that those methods do indeed get called (`expects()` in ScalaMock). I can take a closer look at the unit tests and see if I can come up with something that makes a bit more sense to me. I like the tests that you have (esp if they make sense to you :) ), but personally I feel like they should be marked as integration tests.